usaspending_api/agency/v2/views/program_activity_list.py

```diff
@@ -0,0 +1,66 @@
+from django.db.models import Sum, F
+from rest_framework.request import Request
+from rest_framework.response import Response
+from typing import Any, List
+from usaspending_api.agency.v2.views.agency_base import AgencyBase
+from usaspending_api.common.cache_decorator import cache_response
+from usaspending_api.common.helpers.generic_helper import get_simple_pagination_metadata
+from usaspending_api.references.models import RefProgramActivity
+
+
+class ProgramActivityList(AgencyBase):
+    """
+    Obtain the list of program activity categories for a specific agency in a
+    single fiscal year based on whether or not that program activity has ever
+    been submitted in File B.
+    """
+
+    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/agency/toptier_code/program_activity.md"
+
+    @cache_response()
+    def post(self, request: Request, *args: Any, **kwargs: Any) -> Response:
```

Comment:
Do we need to support `post` here?  With the exception of the advanced search which has a massive filter list that I'm told exceeds maximum query string of older browsers when fully populated, I had envisioned us moving toward more `get` based requests which is typical for REST-like APIs.  `post` is usually saved for operations that add data.

Reply:
I realize there's an argument to be made for being consistent across our endpoints (I've even made that argument), but I feel like this is our opportunity for a clean break.  This is a whole new app.  We can start a new tradition.  One we can call our own.  One that tomorrow's developers can complain about.  "Why'd they do it this way?!?!?!"

Reply:
If there is no pushback I am always in favor of more RESTful endpoints. As you stated, this is purely done as an attempt to be consistent.